src/main/java/com/jwplayer/southpaw/state/RocksDBState.java

```diff
@@ -498,22 +524,17 @@ public void delete(String keySpace, byte[] key) {
     public void deleteBackups() {
         logger.info("Deleting RocksDB state backups");
         try {
-            File file;
-            switch(backupURI.getScheme().toLowerCase()) {
-                case FileHelper.SCHEME:
-                    file = new File(backupURI);
-                    FileUtils.deleteDirectory(file);
-                    break;
-                case S3Helper.SCHEME:
-                    String localBackupPath = getLocalBackupPath(uri);
-                    file = new File(localBackupPath);
-                    FileUtils.deleteDirectory(file);
-                    s3Helper.deleteKeys(backupURI);
-                    break;
-                default:
-                    throw new RuntimeException("Unsupported schema: " + backupURI.getScheme());
+            openBackupEngine();
+            final List<BackupInfo> backupInfoList = backupEngine.getBackupInfo();
```

Comment:
within `openBackupEngine` you guard against null but then immediately call `getBackupInfo`

Reply:
within `openBackupEngine` if `backupEngine` is null it opens it. Otherwise it should already be opened and ready to use. Immediately calling `getBackupInfo` should be fine here